### DISCUSSION TITLE
Cannot switch chain using mobile browsers

### DISCUSSION THREAD

**Discussion by Markcode1992** [wevm/wagmi#4205] [General]
Does anyone else have issues with chain switching? When I use a mobile browser and Metamask I cannot switch chains. Sometimes the first time works and sometimes it doesn't. Also often when it does work I get an error: error: missing or invalid. emit() chainId: eip155:56

I have tried multiple webapps using this wagmi and it happened on all of them. 

In my case I am using useSwitchChain

**Comment by flymauntains** [comment]
I need wagmi version. V1 or V2?

**Reply by Markcode1992** [reply]
This issue is on V2

**Reply by flymauntains** [reply]
Note `https://wagmi.sh/react/api/hooks/useSwitchChain`
